scripts/semonitor.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+set -eu
+if [ -n "${DEBUG_TRACE_SH:-}" ] && \
+   [ "${DEBUG_TRACE_SH:-}" != "${DEBUG_TRACE_SH#*"$(basename "${0}")"*}" ] || \
+   [ "${DEBUG_TRACE_SH:-}" = 'all' ]; then
+	set -x
+fi
+
+se_date_fmt="${SE_DATE_FMT:-%Y%m%d}"
+se_start_year="${SE_START_YEAR:-2023}"
+
+# Wait for a reasonable date to be set
```

Comment:
whats the use case here? is this generally useful?

Reply:
Your dad started to do this in the `scripts/semonitor` script; and I kind of agree this being a good thing. Most of the time, this will be a no-op. But we know from the issues, that sometimes the system comes up with an invalid date, which will be propagated to the inverter. Best to catch that. I did't check for 1969 though; as to be honest, we know that we are beyond 'date of version' so to speak. So checking for 2023 is fine. For those odd use-cases where someone wants to set the date in the past; we got that covered too.

Reply:
I guess it doesn't hurt.  I was more thinking it might be more appropriate to have this check upstream in whatever daemon/service management the system might be using

Reply:
But that puts a lot of responsibility on the host. semonitor is the 'gatekeeper' what is being sent to the inverter. Obviously semonitor has no idea what the correct time is either, but it does know it must be within 2023 or later. We could make it _really_ fancy and try to get UTC time from the internet somewhere, but that breaks offline behavior, and imo that is something rightfully the host should care about.